src/idl_gen_cpp.cpp

```diff
@@ -878,6 +870,18 @@ class CppGenerator : public BaseGenerator {
         code += name + "Identifier()); }\n\n";
       }
 
+      // The root verifier:
+      code += "inline bool Verify";
+      code += name;
+      code +=
+          "Buffer(flatbuffers::Verifier &verifier) { "
+          "return ";
+      if (parser_.file_identifier_.length())
+          code += "verifier.GetBufferSize() > (2 * sizeof(flatbuffers::uoffset_t)) && " + name + "BufferHasIdentifier(verifier.GetBuffer()) && ";
```

Comment:
Ah, I didn't know about generate_code.sh script, I'll run it and push all the changes.